### ISSUE TITLE
Label focus & tab focus issue

### ISSUE THREAD

**Issue by Fewald** [sergiodlopes/jquery-flexdatalist#111]
Hello! 
Thanks for the amazing job, it really saved my time.

Bug: I've met the following issue: when I'm clicking on my "flexdatalist" input, sometimes the page jumps to the top of the page (my page is long, so the form placed under the scroll). 
The same happens when I try to focus my "flexdatalist" input via the tab key. (The cursor on the previous form element, I'm pressing the tab key. Expected: the "flexdatalist" input will be focused; In fact: the page jumps to top;

Reason: I've found the reason for such behavior: the library makes my input itself invisible and exchanges it with a copy of my input called #myinput-flexdatalist. It's fine but my label still pointing on the original input and  that element still in the form tabindex queue. That's why the page "jumps": the original element hides with a "style" value (position: absolute; top: -14000px; left: -14000px;)  which means that this element is very on top and very on left. So the jump is actually just a focus on that.

Fix: I've been able to fix it via not-so-shinning solution: to set dynamically the label "for" value and element "tabindex=-1" value after the flexdatalist initiation. However, the investigation takes time, so I suggest to do something. 

Best,
Sergey

**Comment by sergiodlopes** [comment]
Hi,

Great! Thank you for checking out a fix for the problem. I'll implement a fix on my next release (updating label[for] attribute). 

Thanks!
Sérgio

**Comment by Fewald** [comment]
@sergiodlopes nice, thanks. The second fix is a pretty simple too: AFTER cloning the input, you want to add "tabindex=-1" property to an original element. It guarantees that the original element will not be focused via tab key or iPhone keyboard next/prev buttons.

**Comment by sergiodlopes** [comment]
@Fewald Yeah, that also, although I forgot to mention it in the message... hehe. :)
